test/functional/lua/ssh_spec.lua

```diff
@@ -0,0 +1,44 @@
+local t = require('test.testutil')
+local parser = require('vim._ssh.parser')
+local eq = t.eq
+
+describe('SSH parser', function()
+  it('parses SSH configuration strings', function()
```

Comment:
also need `it('failure modes')` which tests bad or unexpected input, i.e it should use `pcall_err` or similar to assert what happens when parsing fails.


that is almost always required when adding new features.